File: examples/comparative/examples/16/person.py
#!/usr/bin/env python3

class Person:
  __rol = 'jedi'

  def __init__(self, name):
    self.name = name

  def hello1(self):
    print("Hello1:", self.name)

  def hello2(self):
    self.__hello3()

  def __hello3(self):
    print("Hello3:", self.name, "(", Person.__rol, ")")

yoda = Person("Yoda")
yoda.hello1()
yoda.hello2()
print(str(dir(yoda)))
yoda._Person__hello3()
print(yoda._Person__rol)

File: examples/comparative/examples/16/person.rb
#!/usr/bin/env ruby

class Person
  @@rol = "jedi"

  def initialize(name)
    @name = name
  end

  def hello1
    puts "Hello1:" + @name
  end

  def hello2
    hello3
  end

  private

  def hello3
    puts "Hello3:" + @name
    puts "Hello3:" + @@rol
  end
end

obiwan = Person.new("Obiwan")
obiwan.hello1
obiwan.hello2

